### PR TITLE
YDA-5727: Script to update schema-id

### PR DESCRIPTION
**Pull request by claravox**
To run, for example:
`irule -r irods_rule_engine_plugin-python-instance -F tools/update-schema-id.r '*defaultSchema="'default-3'"'`

Tested locally by first deleting schema_id from some groups, and running this script to add them back.

### CODE REVIEW

File: tools/update-schema-id.r [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    userIter = genquery.row_iterator(
+        "USER_GROUP_NAME",
+        "USER_TYPE = 'rodsgroup' AND USER_ZONE = '{}'".format(zone),
+        genquery.AS_LIST,
+        callback) 
+
+    for row in userIter:
+        name = row[0]
+        # Normalish groups
+        if name.startswith("research-") or name.startswith("deposit-") or name.startswith("grp-") or name.startswith("intake-"):
```
Comment (claravox) [inline tools/update-schema-id.r]: Does this get most of the groups we need?

Comment (lwesterhof) [reply tools/update-schema-id.r]: Yeah, but we can limit it to research. The grp group is legacy, intake does not use schemas and deposit is only one group in production.

File: tools/update-schema-id.r [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            schemaIdFound = False
+            for row1 in metaIter:
+                attr = row1[0]
+                if attr == 'schema_id':
+                    schemaIdFound = True
+            
+            if not schemaIdFound:
+                schema_collection = get_schema_collection(callback, zone, name, default_schema)
+                callback.uuGroupModify(name, 'schema_id', schema_collection, '', '') 
+                callback.writeLine("stdout", "Group {} set with schema id {}".format(name, schema_collection))
```
Comment (lwesterhof) [inline tools/update-schema-id.r]: Can we add a couple of extra log statements to show the start and finish of the process and also log groups that do have a schema id and don't need a change?

**Review by lwesterhof — CHANGES_REQUESTED** [review]
Looks good, only some extra logging for the admin running the script would be nice.

**Pull request merged**